types/pluck.d.ts

```diff
@@ -1,10 +1,10 @@
 import { Placeholder } from 'ramda';
 
 export function pluck<K extends PropertyKey>(prop: K extends Placeholder ? never : K): {
-  <U extends readonly unknown[] | Record<K, any>>(obj: Record<PropertyKey, U>): U extends readonly (infer T)[] ? T[] : U extends Record<K, infer T> ? T[] : never;
+  <U extends Record<K, any>, IK extends string>(obj: Record<IK, U>): { [KK in keyof typeof obj]: U[K] };
   <U extends readonly unknown[] | Record<K, any>>(list: U[]): U extends readonly (infer T)[] ? T[] : U extends Record<K, infer T> ? T[] : never;
```

Comment:
I believe you will also need to apply `U extends readonly (infer T)[] ? T[] : U extends Record<K, infer T> ? T[]` that is here to your updated versions for when `obj: Record<>`. This would cover the case of `{ a: string[]; b: string[]; }` or in other works, `Record<string, string[]>`